sklearn/svm/base.py

```diff
@@ -383,12 +395,27 @@ def decision_function(self, X):
             kernel=kernel, degree=self.degree, cache_size=self.cache_size,
             coef0=self.coef0, gamma=self._gamma)
 
-        # In binary case, we need to flip the sign of coef, intercept and
-        # decision function.
-        if self._impl in ['c_svc', 'nu_svc'] and len(self.classes_) == 2:
-            return -dec_func.ravel()
+    def _sparse_decision_function(self, X):
+        X.data = np.asarray(X.data, dtype=np.float64, order='C')
 
-        return dec_func
+        kernel = self.kernel
+        if hasattr(kernel, '__call__'):
+            kernel = 'precomputed'
+
+        kernel_type = self._sparse_kernels.index(kernel)
+
+        return libsvm_sparse.libsvm_sparse_decision_function(
+            X.data, X.indices, X.indptr,
+            self.support_vectors_.data,
+            self.support_vectors_.indices,
+            self.support_vectors_.indptr,
+            self.dual_coef_.data, self._intercept_,
```

Comment:
In accordance with current implementation, you should use [`self._dual_coef_`](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/svm/base.py#L186), since `dual_coef_` may have its sign inverted in a binary case. I suppose, that's the problem Travis encountered.